appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -90,6 +91,55 @@
 		return CryptoSession.execute(password, () -> createResponse(userService.addUser(user), true, HttpStatus.CREATED));
 	}
 
+	private void CheckPassword(Optional<String> password, UUID id)
+	{
+		CryptoSession.execute(password, () -> userService.canAccessPrivateData(id), () -> null);
+	}
+
+	@RequestMapping(value = "{id}/newDeviceRequest", method = RequestMethod.PUT)
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	public HttpEntity<NewDeviceRequestResource> setNewDeviceRequestForUser(
+			@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password, @PathVariable UUID id,
+			@RequestBody CreateNewDeviceRequestDTO createNewDeviceRequest)
+	{
+		CheckPassword(password, id);
+		return createNewDeviceRequestResponse(userService.setNewDeviceRequestForUser(id, password.get(),
+				createNewDeviceRequest.getUserSecret()));
+	}
+
+	@RequestMapping(value = "{id}/newDeviceRequest", params = { "userSecret" }, method = RequestMethod.GET)
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	public HttpEntity<NewDeviceRequestResource> getNewDeviceRequestForUser(@PathVariable UUID id,
+			@RequestParam(value = "userSecret") String userSecret)
+	{
+		return createNewDeviceRequestResponse(userService.getNewDeviceRequestForUser(id, userSecret));
+	}
+
+	@RequestMapping(value = "{id}/newDeviceRequest", method = RequestMethod.DELETE)
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	public void clearNewDeviceRequestForUser(@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password,
+			@PathVariable UUID id)
+	{
+		CheckPassword(password, id);
+		userService.clearNewDeviceRequestForUser(id);
+	}
+
+	private HttpEntity<NewDeviceRequestResource> createNewDeviceRequestResponse(NewDeviceRequestDTO newDeviceRequest)
+	{
+		return new ResponseEntity<NewDeviceRequestResource>(new NewDeviceRequestResource(newDeviceRequest), HttpStatus.OK);
+	}
+
+	public static class NewDeviceRequestResource extends Resource<NewDeviceRequestDTO>
+	{
+		public NewDeviceRequestResource(NewDeviceRequestDTO newDeviceRequest)
+		{
+			super(newDeviceRequest);
+		}
+	}
+
```

Comment:
Done, but you don't need a resource assembler, only changed the resource constructor.